ansible/playbooks/roles/grafana/tasks/upgrade/main.yml

```
@@ -0,0 +1,97 @@
+---
+- name: Include new version of defaults
+  include_vars:
+    file: defaults/versions.yml
+    name: grafana_versions
+
+- name: Include upgrade defaults
+  include_vars:
+    file: defaults/upgrade.yml
+    name: upgrade_defaults
+
+- name: Populate service facts
+  service_facts:
+  when: ansible_facts.service is undefined
```

```suggestion
- name: Get information about installed packages as facts
  package_facts:
    manager: auto
  when: ansible_facts.packages is undefined
```
Grafana is installed as package so IMO it should be enough just to check whether it exists (without checking binary file and service - so we could remove `Check Grafana binary existance` task as well).